runtime/src/append_vec.rs

```diff
@@ -899,16 +899,16 @@ pub mod tests {
         let accounts = av.accounts(0);
         let account = accounts.first().unwrap();
 
+        // upper 7-bits are not 0, so sanitization should fail
+        assert!(!account.sanitize_executable());
```

Comment:
:+1:

my memory is vague. but i think i tried to add this assertion, but i couldn't because the ub assertion below stopped working correctly back then. (welcome to UB land. lol)

if today's build toolchain doesn't negatively affect the UB detection code, please leave to `assert!` here.

Reply:
^passing CI with the assert left there.